File: CSharp/Eta/Src/Core/Codec/ClassesOfService.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license      --
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.  --
 *|                See the project's LICENSE.md for details.                  --
 *|           Copyright (C) 2022-2023 Refinitiv. All rights reserved.              --
 *|-----------------------------------------------------------------------------
 */

using LSEG.Eta.Codec;

namespace LSEG.Eta.Rdm
{
	/// <summary>
	/// Qualified stream class of service specific RDM definitions. </summary>
	sealed public class ClassesOfService
	{
		// ClassesOfService class cannot be instantiated
		private ClassesOfService()
		{
            throw new System.NotImplementedException();
        }

		/// <summary>
		/// Filter IDs for class of service.
		/// </summary>
		public class FilterIds
		{
			// FilterIds class cannot be instantiated
			private FilterIds()
			{
                throw new System.NotImplementedException();
			}

			/// <summary>
			/// Common Properties Filter ID </summary>
			public const int COMMON_PROPERTIES = 1;
			/// <summary>
			/// Authentication Filter ID </summary>
			public const int AUTHENTICATION = 2;
			/// <summary>
			/// Flow Control Filter ID </summary>
			public const int FLOW_CONTROL = 3;
			/// <summary>
			/// Data Integrity Filter ID </summary>
			public const int DATA_INTEGRITY = 4;
			/// <summary>
			/// Guarantee Filter ID </summary>
			public const int GUARANTEE = 5;
		}

		/// <summary>
		/// Filter flags for class of service.
		/// </summary>
		public class FilterFlags
		{
			// FilterFlags class cannot be instantiated
			internal FilterFlags()
			{
                throw new System.NotImplementedException();
            }

			/// <summary>
			/// Common Properties Filter Mask </summary>
			public const int COMMON_PROPERTIES = 0x00000001;
			/// <summary>
			/// Authentication Filter Mask </summary>
			public const int AUTHENTICATION = 0x00000002;
			/// <summary>
			/// Flow Control Filter Mask </summary>
			public const int FLOW_CONTROL = 0x00000004;
			/// <summary>
			/// Data Integrity Filter Mask </summary>
			public const int DATA_INTEGRITY = 0x00000008;
			/// <summary>
			/// Guarantee Filter Mask </summary>
			public const int GUARANTEE = 0x00000010;
		}

		/// <summary>
		/// Element names for class of service.
		/// </summary>
		public class ElementNames
		{
			// ElementNames class cannot be instantiated
			internal ElementNames()
			{
                throw new System.NotImplementedException();
            }

			/// <summary>
			/// :StreamVersion </summary>
			public static readonly Buffer STREAM_VERSION = new Buffer();

			/// <summary>
			/// :MaxMsgSize </summary>
			public static readonly Buffer MAX_MSG_SIZE = new Buffer();

			/// <summary>
			/// :ProtocolType </summary>
			public static readonly Buffer PROTOCOL_TYPE = new Buffer();

			/// <summary>
			/// :ProtocolMajorVersion </summary>
			public static readonly Buffer PROTOCOL_MAJOR_VERSION = new Buffer();

			/// <summary>
			/// :ProtocolMinorVersion </summary>
			public static readonly Buffer PROTOCOL_MINOR_VERSION = new Buffer();

			/// <summary>
			/// :RecvWindowSize </summary>
			public static readonly Buffer RECV_WINDOW_SIZE = new Buffer();

			/// <summary>
			/// :Type </summary>
			public static readonly Buffer TYPE = new Buffer();

            /// <summary>
            /// 
            /// </summary>
			static ElementNames()
            {
				// :StreamVersion
				STREAM_VERSION.Data(":StreamVersion");

				// :MaxMsgSize
				MAX_MSG_SIZE.Data(":MaxMsgSize");

                // :ProtocolType
                PROTOCOL_TYPE.Data(":ProtocolType");

				// :ProtocolMajorVersion
				PROTOCOL_MAJOR_VERSION.Data(":ProtocolMajorVersion");

				// :ProtocolMinorVersion </summary>
				PROTOCOL_MINOR_VERSION.Data(":ProtocolMinorVersion");

				// :RecvWindowSize
				RECV_WINDOW_SIZE.Data(":RecvWindowSize");

				// :Type </summary>
				TYPE.Data(":Type");
			}
		}

		/// <summary>
		/// Authentication type enumerations for class of service.
		/// </summary>
		public class AuthenticationTypes
		{
			// AuthenticationTypes class cannot be instantiated
			private AuthenticationTypes()
			{
                throw new System.NotImplementedException();
            }

			/// <summary>
            /// Not required
            /// </summary>
			public const int NOT_REQUIRED = 0;

			/// <summary>
            /// OMM Login
            /// </summary>
			public const int OMM_LOGIN = 1;
		}

		/// <summary>
		/// Flow control type enumerations for class of service.
		/// </summary>
		public class FlowControlTypes
		{
			// FlowControlTypes class cannot be instantiated
			private FlowControlTypes()
			{
                throw new System.NotImplementedException();
            }

			/// <summary>
			/// None </summary>
			public const int NONE = 0;

			/// <summary>
			/// Bidirectional </summary>
			public const int BIDIRECTIONAL = 1;
		}

		/// <summary>
		/// Data integrity type enumerations for class of service.
		/// </summary>
		public class DataIntegrityTypes
		{
			// DataIntegrityTypes class cannot be instantiated
			private DataIntegrityTypes()
			{
                throw new System.NotImplementedException();
            }

			/// <summary>
			/// Best Effort </summary>
			public const int BEST_EFFORT = 0;

			/// <summary>
			/// Reliable </summary>
			public const int RELIABLE = 1;
		}

		/// <summary>
		/// Guarantee type enumerations for class of service.
		/// </summary>
		public class GuaranteeTypes
		{
			// Guarantee classTypes cannot be instantiated
			private GuaranteeTypes()
			{
                throw new System.NotImplementedException();
            }

			/// <summary>
			/// None </summary>
			public const int NONE = 0;

			/// <summary>
			/// PersistentQueue </summary>
			public const int PERSISTENT_QUEUE = 1;
		}
	}

}

File: Java/Eta/Core/src/main/java/com/refinitiv/eta/rdm/ClassesOfService.java
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2020,2025 LSEG. All rights reserved.
 *|-----------------------------------------------------------------------------
 */

package com.refinitiv.eta.rdm;

import com.refinitiv.eta.codec.Buffer;
import com.refinitiv.eta.codec.CodecFactory;

/** Qualified stream class of service specific RDM definitions. */
public class ClassesOfService
{
    // ClassesOfService class cannot be instantiated
    private ClassesOfService()
    {
        throw new AssertionError();
    }

    /**
     * Filter IDs for class of service.
     */
    public static class FilterIds
    {
        // FilterIds class cannot be instantiated
        private FilterIds()
        {
            throw new AssertionError();
        }

        /** Common Properties Filter ID */
        public static final int COMMON_PROPERTIES = 1;
        /** Authentication Filter ID */
        public static final int AUTHENTICATION = 2;
        /** Flow Control Filter ID */
        public static final int FLOW_CONTROL = 3;
        /** Data Integrity Filter ID */
        public static final int DATA_INTEGRITY = 4;
        /** Guarantee Filter ID */
        public static final int GUARANTEE = 5;
    }

    /**
     * Filter flags for class of service.
     */
    public static class FilterFlags
    {
        // FilterFlags class cannot be instantiated
        private FilterFlags()
        {
            throw new AssertionError();
        }

        /** Common Properties Filter Mask */
        public static final int COMMON_PROPERTIES = 0x00000001;
        /** Authentication Filter Mask */
        public static final int AUTHENTICATION = 0x00000002;
        /** Flow Control Filter Mask */
        public static final int FLOW_CONTROL = 0x00000004;
        /** Data Integrity Filter Mask */
        public static final int DATA_INTEGRITY = 0x00000008;
        /** Guarantee Filter Mask */
        public static final int GUARANTEE = 0x00000010;
    }

    /**
     * Element names for class of service.
     */
    public static class ElementNames
    {
        // ElementNames class cannot be instantiated
        private ElementNames()
        {
            throw new AssertionError();
        }
        
        /** :StreamVersion */
        public static final Buffer STREAM_VERSION = CodecFactory.createBuffer();        

        /** :MaxMsgSize */
        public static final Buffer MAX_MSG_SIZE = CodecFactory.createBuffer();        

        /** :MaxFragmentSize */
        public static final Buffer MAX_FRAGMENT_SIZE = CodecFactory.createBuffer();        
        
        /** :ProtocolType */
        public static final Buffer PROTOCOL_TYPE = CodecFactory.createBuffer();        

        /** :ProtocolMajorVersion */
        public static final Buffer PROTOCOL_MAJOR_VERSION = CodecFactory.createBuffer();        

        /** :ProtocolMinorVersion */
        public static final Buffer PROTOCOL_MINOR_VERSION = CodecFactory.createBuffer();        

        /** :RecvWindowSize */
        public static final Buffer RECV_WINDOW_SIZE = CodecFactory.createBuffer();        
        
        /** :Type */
        public static final Buffer TYPE = CodecFactory.createBuffer();
        
        /** :SupportsFragmentation */
        public static final Buffer SUPPS_FRAGMENTATION = CodecFactory.createBuffer();      
        
        static
        {
            /** :StreamVersion */
            STREAM_VERSION.data(":StreamVersion");
            
            /** :MaxMsgSize */
            MAX_MSG_SIZE.data(":MaxMsgSize");
            
            /** :MaxFragmentSize */
            MAX_FRAGMENT_SIZE.data(":MaxFragmentSize");            
            
            /** :ProtocolType */
            PROTOCOL_TYPE.data(":ProtocolType");
            
            /** :ProtocolMajorVersion */
            PROTOCOL_MAJOR_VERSION.data(":ProtocolMajorVersion");
            
            /** :ProtocolMinorVersion */
            PROTOCOL_MINOR_VERSION.data(":ProtocolMinorVersion");
            
            /** :RecvWindowSize */
            RECV_WINDOW_SIZE.data(":RecvWindowSize");
            
            /** :Type */
            TYPE.data(":Type");
            
            /** :SupportsFragmentation */
            SUPPS_FRAGMENTATION.data(":SupportsFragmentation");
        }
    }

    /**
     * Authentication type enumerations for class of service.
     */
    public static class AuthenticationTypes
    {
        // AuthenticationTypes class cannot be instantiated
        private AuthenticationTypes()
        {
            throw new AssertionError();
        }

        /** Not Required */
        public static final int NOT_REQUIRED = 0;

        /** OMM Login */
        public static final int OMM_LOGIN = 1;
    }

    /**
     * Flow control type enumerations for class of service.
     */
    public static class FlowControlTypes
    {
        // FlowControlTypes class cannot be instantiated
        private FlowControlTypes()
        {
            throw new AssertionError();
        }

        /** None */
        public static final int NONE = 0;

        /** Bidirectional */
        public static final int BIDIRECTIONAL = 1;
    }

    /**
     * Data integrity type enumerations for class of service.
     */
    public static class DataIntegrityTypes
    {
        // DataIntegrityTypes class cannot be instantiated
        private DataIntegrityTypes()
        {
            throw new AssertionError();
        }

        /** Best Effort */
        public static final int BEST_EFFORT = 0;

        /** Reliable */
        public static final int RELIABLE = 1;
    }

    /**
     * Guarantee type enumerations for class of service.
     */
    public static class GuaranteeTypes
    {
        // Guarantee classTypes cannot be instantiated
        private GuaranteeTypes()
        {
            throw new AssertionError();
        }

        /** None */
        public static final int NONE = 0;

        /** PersistentQueue */
        public static final int PERSISTENT_QUEUE = 1;
    }
}
